src/SelectPicker/test/SelectPickerSpec.js

```diff
@@ -400,4 +400,24 @@ describe('SelectPicker', () => {
       expect(getByTestId('content')).to.have.text('Not selected');
     });
   });
+  it('SearchWord should be reset when controlled and triggered off', async () => {
+    const handleSearch = sinon.spy();
+    const Wrapper = () => {
+      return (
+        <>
+          <button id="exit">exit</button>
+          <SelectPicker container={container} defaultOpen onSearch={handleSearch} data={data} />
+        </>
+      );
+    };
+    Wrapper.displayName = 'WrapperSelectPicker';
+    let { container } = render(<Wrapper />);
+    const exit = container.querySelector('#exit');
+
+    // close select
+    fireEvent.mouseDown(exit, { bubbles: true });
+    await waitForElementToBeRemoved(document.querySelector('.rs-picker-search-bar-input'));
+
+    assert.isTrue(handleSearch.calledOnce);
```

Comment:
Avoid mentioning implementation detail in test cases. Simply use `waitFor` from `@testing-library/react` for async assertions.

---

Arguments of onSearch call should also be asserted.

Reply:
Yeah